equinix/resource_metal_device.go

```diff
@@ -607,9 +612,9 @@ func resourceMetalDeviceCreate(ctx context.Context, d *schema.ResourceData, meta
 
 func resourceMetalDeviceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) error {
 	meta.(*Config).addModuleToMetalUserAgent(d)
-	client := meta.(*Config).metal
+	client := meta.(*Config).metalgo
```

Comment:
I haven't converted the create/update/delete functions yet; we could _potentially_ migrate this resource (and resources in general) from `packngo` to `metal-go` one function at a time in order to have smaller PRs that are easier to review.  The major complication with that is if we have tests that expect a specific SDK to be used, but that complication is already mentioned elsewhere in this PR.

Reply:
After the conversion, I vote we strip out the "Go" in these function names

Reply:
Yeah, the suffix is only here to keep the code paths separate and should be removed once it's no longer needed.